locust_tasks/locustfile.py

```diff
@@ -166,49 +189,163 @@ def correct_address(self):
+            verify_response('RequestUacPost-Start', self, response, 200, Page.START)
+
+    @task(2)
+    def enter_postcode(self):
+        """
+        POST postcode
+        """
+        with self.client.post("/en/requests/access-code/enter-address/", {
+            'form-enter-address-postcode': self.case['postcode']
+        }, catch_response=True) as response:
+            self.address_to_select = extractAddress(response, self.case["uprn"])
+            verify_response('RequestUacPost-EnterAddress', self, response, 200, Page.SELECT_ADDRESS,
+                            self.case["postcode"])
+
+    @task(3)
+    def select_address(self):
+        """
+        POST uprn and whole address extracted as JSON from the HTML in previous task
+        """
+        logger.info("Address selected: " + self.address_to_select)
+        with self.client.post("/en/requests/access-code/select-address/", {
+            'form-select-address': self.address_to_select
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-SelectAddress', self, response, 200, Page.ADDRESS_CORRECT, self.case["addressLine1"])
+
+    @task(4)
+    def confirm_address(self):
+        """
+        POST 'yes' to confirm address
+        """
+        with self.client.post("/en/requests/access-code/confirm-address/", {
+            'form-confirm-address': 'yes'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-ConfirmAddress', self, response, 200, Page.SELECT_METHOD, "Post")
+
+    @task(5)
+    def select_method(self):
+        """
+        POST 'post' to select post as method of sending UACs
+        """
+        with self.client.post("/en/requests/access-code/select-method/", {
+            'form-select-method': 'post'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-SelectMethod', self, response, 200, Page.ENTER_NAME, "")
+
+    @task(6)
+    def enter_name(self):
+        """
+        POST 'John' as first name and 'Smith' as last name of person to send the UAC to
+        """
+        with self.client.post("/en/requests/access-code/enter-name/", {
+            'name_first_name': 'John',
+            'name_last_name': 'Smith'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-EnterName', self, response, 200, Page.CONFIRM_NAME,
+                            "John Smith<br>")
 
     @task(7)
-    def start_for_new_uac(self):
-        self.client.get("/en/start/")
+    def confirm_name_address(self):
+        """
+        POST 'yes' to confirm name and address
+        """
+        with self.client.post("/en/requests/access-code/confirm-name-address/", {
+            'request-name-address-confirmation': 'yes'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-ConfirmName', self, response, 200, Page.CODE_SENT, "John Smith")
```

Comment:
Following on from the previous comment, this would need check that the page contained the previously generated random name.

Reply:
makes sense

Reply:
I've added a TODO for after CR-1379 is done